### PR TITLE
Add files for basic ci (lint, test, coverage)

### PR DESCRIPTION
**Pull request by cescgina**
Update tox and add pre-commit hook for more complete testing and linting

### CODE REVIEW

**Comment by cescgina** [comment]
Now we have the CI running. It is not passing due to many issues with pylint and no unit tests, but I think that this PR is big enough as it is and it would be better to create a different one with fixes and more tests.

File: znoyder/cli.py [hunk 1]
```diff
@@ -16,17 +16,10 @@
...
-
-from znoyder import browser
-from znoyder import downloader
-from znoyder import finder
-from znoyder import generator
-from znoyder import templater
+from argparse import (REMAINDER, ArgumentParser, Namespace,
+                      RawDescriptionHelpFormatter)
 
+from znoyder import browser, downloader, finder, generator, templater
```
Comment (sdatko) [inline znoyder/cli.py:22]: Hmmm, AFAIK one import per line is preferred to multiple imports in single line. Why do we change that?

File: tox.ini [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+skip_missing_interpreters = False
+requires =
+    tox-extra; python_version >= '3.8'
 
 [testenv]
-basepython = python3
-usedevelop = False
-install_command = pip install {opts} {packages}
-
-[testenv:run]
```
Comment (sdatko) [inline tox.ini:11]: Please lets keep the run env, I use it sometimes just to execute some znoyder command without venv

Comment (cescgina) [reply tox.ini:11]: Sure, I'll add it back

File: znoyder/cli.py (hunk 1 above)
Comment (cescgina) [reply znoyder/cli.py:22]: That was changed automatically by isort, which is part of the linters used by pre-commit. I used the same config that we have setup in cibyl, I thought I would help to have the same standards, but can be adjusted

**Review by sdatko — CHANGES_REQUESTED** [review]
> Now we have the CI running. It is not passing due to many issues with pylint and no unit tests, but I think that this PR is big enough as it is and it would be better to create a different one with fixes and more tests.

So, I agree this PR it big enough. The original task was about introducing envs for unit tests and coverage, while here there is 11 commits, pylint involved and even some syntax changes. We even did not discuss about pylint, which IMO is too un-useful for real work with all its extremist rules for things like variables, docstrings and public methods.

I propose to add here just what was explicitly asked in task and for further proposals (enabling pylint and other linters) lets have another changes.

**Comment by cescgina** [comment]
My bad, I misunderstood the task. In a minute I'll submit a simpler change that should do it

**Review by sdatko — APPROVED** [review]
Thanks, I like it much more now, as it is simpler and exactly to the point.
We can start working on actual unit tests now. Thanks!

Whether we want introduce additional linting or change something, lets raise that on our project sync.

**Pull request merged**